### PR TITLE
Add permissions system

### PR DESCRIPTION
**Pull request by mets11rap**
ALO - Authority Level Oriented

Add authority levels to server roles and restrict commands to minimum authority levels

### CODE REVIEW

File: lib/modules/command-handler/index.js [hunk 1]
```diff
@@ -158,13 +158,18 @@ class CommandHandler {
    * @param   {Error}  [Type=Error] The error constructor class
    * @returns {Error}               The resulting error
    */
-  _generateError (name, message, code, Type = Error) {
-    const err = new Type(`${name} [${code}]: ${message}`)
+  _generateError (name, message, code, Type) {
```
Comment (alex-taxiera) [inline lib/modules/command-handler/index.js:161]: why not using a base class for the error?

File: lib/modules/reaction-handler/index.js [hunk 2]
```diff
@@ -135,13 +136,18 @@ class ReactionHandler {
    * @param   {Error}  [Type=Error] The error constructor class
    * @returns {Error}               The resulting error
    */
-  _generateError (name, message, code, Type = Error) {
-    const err = new Type(`${name} [${code}]: ${message}`)
+  _generateError (name, message, code, Type) {
```
Comment (alex-taxiera) [inline lib/modules/reaction-handler/index.js:139]: duplicated code with other error generator(s)

File: lib/structures/command/index.js [hunk 3]
```diff
@@ -41,13 +43,15 @@ class Command {
...
+     * @prop {Boolean}    guildOnly  Whether or not this command can only be triggered in guilds as opposed to DMs
+     * @prop {Number}     authLevel  The minimum auth level required to execute this command
      */
     this.options = {
       args,
       aliases: (Array.isArray(aliases) ? aliases : [aliases]).map((a) => a.toLowerCase()),
       restricted,
-      serverOnly
+      guildOnly,
+      authLevel: parseInt(authLevel) || 0
```
Comment (alex-taxiera) [inline lib/structures/command/index.js:54]: what happens if authLevel is NaN

Comment (exoRift) [reply lib/structures/command/index.js:54]: NaN is falsy

**Pull request merged**